Cython/Compiler/TypeSlots.py

```diff
@@ -1021,7 +1021,7 @@ def get_slot_code_by_name(scope, slot_name):
     EmptySlot("tp_weaklist"),
     EmptySlot("tp_del"),
     EmptySlot("tp_version_tag"),
-    DelSlot("tp_finalize", ["__del__"], "0", ifdef="PY_VERSION_HEX >= 0x030400a1"),
+    DelSlot("tp_finalize", ["__del__"], "0", ifdef="PY_VERSION_HEX >= 0x030400a1 && CYTHON_USE_TP_FINALIZE"),
```

Comment:
I'm don't think that's right. The version I suggested sets the slot to NULL when `CYTHON_USE_TP_FINALIZE` isn't set while your version assumes that the slot doesn't exist.

I admit that my version is really ugly though and there should probably be a better way to do it.